benchmark/src/main/java/com/linecorp/decaton/benchmark/AsyncProfilerProfiling.java

```diff
@@ -0,0 +1,114 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Profiler for execution.
+ * Internally uses async-profiler https://github.com/jvm-profiling-tools/async-profiler to get
+ * stack samples.
+ */
+@Slf4j
+public class AsyncProfilerProfiling implements Profiling {
+    private static final int PROFILER_CMD_TIMEOUT_SECS = 30;
+
+    private final Path asyncProfilerBin;
+    private final List<String> asyncProfilerOpts;
+
+    public AsyncProfilerProfiling(Path asyncProfilerBin, Collection<String> asyncProfilerOpts) {
+        this.asyncProfilerBin = asyncProfilerBin;
+        if (asyncProfilerOpts == null) {
+            this.asyncProfilerOpts = new ArrayList<>();
+        } else {
+            this.asyncProfilerOpts = new ArrayList<>(asyncProfilerOpts);
+        }
+        if (!this.asyncProfilerOpts.contains("-f")) {
+            this.asyncProfilerOpts.add("-f");
+            this.asyncProfilerOpts.add(outputFileName());
+        }
+    }
+
+    private static long currentPid() {
+        String[] names = ManagementFactory.getRuntimeMXBean().getName().split("@", 2);
+        return Long.parseLong(names[0]);
+    }
+
+    private static String outputFileName() {
+        return "profile-" + currentPid() + ".svg";
+    }
+
+    private void exec(String subCommand) {
+        List<String> cmd = new ArrayList<>();
+        cmd.add(asyncProfilerBin.toString());
+        cmd.addAll(asyncProfilerOpts);
+        cmd.add(subCommand);
+        cmd.add(String.valueOf(currentPid()));
+        try {
+            Process process = new ProcessBuilder(cmd)
+                    .redirectOutput(new File("/dev/stderr"))
+                    .redirectError(Redirect.INHERIT)
+                    .start();
+            if (!process.waitFor(PROFILER_CMD_TIMEOUT_SECS, TimeUnit.SECONDS)) {
+                throw new RuntimeException("timed out waiting async-profiler command");
+            }
+            if (process.exitValue() != 0) {
+                throw new RuntimeException("async-profiler exits with error: " + process.exitValue());
+            }
+        } catch (Exception e) {
+            log.error("Failed to run profiler command: {}", cmd, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void start() {
+        log.info("Start profiling execution");
+        exec("start");
+    }
+
+    private Optional<Path> findOutputPath() {
+        for (int i = 0; i < asyncProfilerOpts.size() - 1; i++) {
+            if ("-f".equals(asyncProfilerOpts.get(i))) {
+                // Avoid printing absolute path expecting the result to be copy & pasted in public
+                // place by security concerns.
+                Path path = Paths.get("").toAbsolutePath().relativize(
+                        Paths.get("", asyncProfilerOpts.get(i + 1)));
```

Comment:
That's true, shame I missed it in testing locally ;;
As the last resort I think I can make the field to have type `String` rather than `Path` so that it won't get converted into fullpath. I'll try that. 

> So instead of relativize here, necessary to construct relative path in BenchmarkResult#print?

I don't want to do this because I want to keep the same security property for (to be added) JSON output.